File: examples/producer/typescript/src/producer/responses/messages.request.ts
import { Response } from "../implementation/responses/messages.request";
import {
	Context,
	Producer,
	Identification,
	Filter,
	Protocol,
} from "../implementation/responses";
import { Scope } from "../implementation/scope";

export function response(
	request: Protocol.Messages.Request,
	scope: Scope
): Promise<Response> {
	return Promise.resolve(
		new Response(
			new Protocol.Messages.Response({
				messages: scope.context.getMessages(),
			})
		)
	);
}

File: examples/producer/rust/src/producer/responses/messages_request.rs
use super::{identification, producer::Control, protocol, scope::Scope, Context};
use clibri::server;

#[allow(unused_variables)]
pub async fn response<E: server::Error, C: server::Control<E>>(
    request: &protocol::Messages::Request,
    scope: &mut Scope<'_, E, C>,
) -> Result<protocol::Messages::Response, protocol::Messages::Err> {
    Ok(protocol::Messages::Response {
        messages: scope.context.get_messages(),
    })
}
